index.js

```diff
@@ -2,94 +2,123 @@ import React, { Component } from 'react';
+        selectedDataArray.push(data[itemsArray[i]]);
       }
 
       return selectedDataArray;
     }
   };
 
-  calculateBoxWidth = maxPerRow => (this.state.width / maxPerRow);
-
   handleSelectItem = (keyValue) => {
-    const { maxSelect } = this.props;
+    const { maxSelect, onSelect } = this.props;
+    const { selectedItem, itemsArray } = this.state;
 
     if (maxSelect === 1) {
-      if (this.state.selectedItem === keyValue) {
-        this.setState({ selectedItem: -1 });
-      } else {
-        this.setState({ selectedItem: keyValue });
-      }
+      this.setState({
+        selectedItem: selectedItem === keyValue ? -1 : keyValue,
+      });
+      onSelect(selectedItem === keyValue ? -1 : keyValue);
     } else if (maxSelect > 1) {
-      const newItemsArray = this.state.itemsArray;
+      const newItemsArray = itemsArray;
 
       // Check if keyValue already exist in array
-      const itemIndex = newItemsArray.length > 0 ? this.getIndex(keyValue, newItemsArray) : -1;
+      const itemIndex =
+        newItemsArray.length > 0 ? this.getIndex(keyValue, newItemsArray) : -1;
 
       // If item found in array
       if (itemIndex > -1) {
         newItemsArray.splice(itemIndex, 1);
-      } else if (this.state.itemsArray.length < maxSelect) {
+      } else if (itemsArray.length < maxSelect) {
         newItemsArray.push(keyValue);
       }
 
       this.setState({ itemsArray: newItemsArray });
+      onSelect(newItemsArray);
     }
-  }
+  };
 
   render() {
-    const { data, maxPerRow, unselectedRender, selectedRender, selectedStyle, unselectedStyle } = this.props;
-    const boxWidth = this.calculateBoxWidth(maxPerRow);
-    let counter = 0;
+    const {
+      data,
+      maxPerRow,
+      unselectedRender,
+      selectedRender,
+      selectedStyle,
+      unselectedStyle,
+      height,
+    } = this.props;
+    const { selectedItem, itemsArray } = this.state;
+
+    const content = [];
+    for (let i = 0, counter = 0; i < data.length / maxPerRow; i++) {
+      const row = [];
+      for (let k = 0; k < maxPerRow; k++) {
+        const keyValue = counter;
+        if (counter < data.length) {
+          row.push(
+            <TouchableOpacity
+              style={[
+                styles.contentBox,
+                unselectedStyle,
+                (selectedItem === keyValue || itemsArray.includes(keyValue)) &&
+                  selectedStyle,
+                height == null ? { aspectRatio: 1 } : { height },
+                {
+                  flex: 1,
+                },
+              ]}
+              onPress={() => this.handleSelectItem(keyValue)}
+            >
+              {selectedItem === keyValue || itemsArray.includes(keyValue)
+                ? selectedRender(data[keyValue])
+                : unselectedRender(data[keyValue])}
+            </TouchableOpacity>,
+          );
+        } else {
+          row.push(
+            <View style={{ flex: 1, backgroundColor: 'transparent' }} />,
```

Comment:
Please add a `key` as each child in an array or iterator should have a unique "key" prop.